### PR TITLE
[RORDEV-707] Create data stream fix, data streams API tests

### PR DESCRIPTION
**Pull request by mateuszkp96**
🐞Fix (ES) Data streams creation issue fix

### CODE REVIEW

File: es710x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/ReflectionBasedIndicesEsRequestContext.scala
Comment (coutoPL) [inline es710x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/ReflectionBasedIndicesEsRequestContext.scala:60]: we should handle all requests from the `org.elasticsearch.action.datastreams` package explicitly. They should be handled before this generic index-based request handling. Let's treat the name of the stream as the index name at the moment. It should be safe enough but it'll unblock the data stream API (currently clients cannot use it because of the bug)

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: es711x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/datastreams/ReflectionBasedDataStreamsEsRequestContext.scala
Comment (coutoPL) [inline es711x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/datastreams/ReflectionBasedDataStreamsEsRequestContext.scala:39]: `methodName` is too generic

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/scala/tech/beshu/ror/integration/plugin/DataStreamApiSuite.scala
Comment (mateuszkp96) [reply integration-tests/src/test/scala/tech/beshu/ror/integration/plugin/DataStreamApiSuite.scala]: Finally, I used `waitForRefresh` and I reduced a number of created documents. With `waitForRefresh`  the response is after 1 second

File: es710x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/datastreams/GetDataStreamEsRequestContext.scala
Comment (coutoPL) [inline es710x/src/main/scala/tech/beshu/ror/es/handler/request/context/types/datastreams/GetDataStreamEsRequestContext.scala:36]: do we need it?

File: integration-tests/src/test/scala/tech/beshu/ror/integration/plugin/DataStreamApiSuite.scala
Comment (coutoPL) [inline integration-tests/src/test/scala/tech/beshu/ror/integration/plugin/DataStreamApiSuite.scala:39]: I see one problem with these tests. We test the API as an admin user which has no indices restriction. It would be better if we can use user1 | user2 | user3. Eg we should not be able to create a datastream that has ".ds-data-stream-dev-0001" index underneath if the user has no proper pattern allowed (eg. .ds-data-stream-dev*). I see you prepared proper ROR config, but these users were not used in this suite.

**Review by coutoPL — CHANGES_REQUESTED** [review]

**Review by coutoPL — APPROVED** [review]
LGTM.

Please add changelog style description to the PR before merging it. 
Thanks

**Review by coutoPL — APPROVED** [review]

**Pull request merged**
